lib/oli/publishing/authoring_resolver.ex

```diff
@@ -285,4 +285,65 @@ defmodule Oli.Publishing.AuthoringResolver do
       numbering_tracker
     }
   end
+
+  @doc """
+  Returns the current revisions of all page resources whose purpose type matches the one it receives as parameter
+  ## Examples
+      iex> get_by_purpose(valid_project_slug, valid_purpose)
+      [%Revision{}, ...]
+
+      iex> get_by_purpose(invalid_project_slug, invalid_purpose)
+      []
+  """
+
+  def get_by_purpose(project_slug, purpose) do
+    Repo.all(
+      from(
+        revision in Revision,
+        left_join: revision2 in Revision,
+        on: revision2.resource_id == revision.resource_id and revision.id < revision2.id,
```

Comment:
This query must instead leverage the "working publication" for the project to ensure it is picking up the most recent revision for this project, and not just the "most recent revision". See the fragment in this file `project_working_publication(project_slug)` and how it is used in other resolver functions. 

This will fail when a project is duplicated from an existing project.  Consider project A that contains page resource B.  If I create a duplicate project C, and in the context of C begin editing page B - there will be later revisions of page B that do not correspond to project A. 

We also should not be hardcoding the resource type id of `1` here.  Instead, use a call to `Oli.Resouurces.ResourceType.get_id_by_type("page")`

Reply:
Good! I understand. I'll fix it